core/src/main/java/com/microsoft/applicationinsights/channel/concrete/localforwarder/LocalForwarderTelemetryChannel.java

```diff
@@ -0,0 +1,102 @@
+package com.microsoft.applicationinsights.channel.concrete.localforwarder;
+
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.microsoft.applicationinsights.channel.concrete.ATelemetryChannel;
+import com.microsoft.applicationinsights.internal.channel.TransmitterFactory;
+import com.microsoft.applicationinsights.internal.channel.common.TelemetryBuffer;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.util.LimitsEnforcer;
+import com.microsoft.applicationinsights.telemetry.BaseTelemetry;
+import com.microsoft.localforwarder.library.inputs.contracts.Telemetry;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import java.util.Map;
+
+public class LocalForwarderTelemetryChannel extends ATelemetryChannel<Telemetry> {
+
+    public static final String ENDPOINT_ENVIRONMENT_VARIABLE_NAME = "APPLICATION_INSIGHTS_LOCAL_FORWARDER_ENDPOINT";
+    public static final String ENDPOINT_SYSTEM_PROPERTY_NAME = "applicationinsights.localforwarder.endpoint";
+
+    public LocalForwarderTelemetryChannel(String endpointAddress, boolean developerMode, int maxTelemetryBufferCapacity, int sendIntervalInMillis) {
+        super(endpointAddress, developerMode, maxTelemetryBufferCapacity, sendIntervalInMillis);
+    }
+
+    public LocalForwarderTelemetryChannel(String endpointAddress, String maxTransmissionStorageCapacity, boolean developerMode, int maxTelemetryBufferCapacity, int sendIntervalInMillis, boolean throttling, int maxInstantRetries) {
+        super(endpointAddress, maxTransmissionStorageCapacity, developerMode, maxTelemetryBufferCapacity, sendIntervalInMillis, throttling, maxInstantRetries);
+    }
+
+    public LocalForwarderTelemetryChannel(Map<String, String> namesAndValues) {
+        super(namesAndValues);
+    }
+
+    @Override
+    protected synchronized void initialize(String configurationFileEndpoint, String maxTransmissionStorageCapacity, boolean developerMode,

```

Comment:
Is there protection against this being called twice?

Reply:
added it